includes/class-wp-job-manager-post-types.php

```diff
@@ -659,6 +660,69 @@ public function job_feed() {
 		remove_filter( 'posts_search', 'get_job_listings_keyword_search' );
 	}
 
+	/**
+	 * Get filled jobs.
+	 *
+	 * @return array
+	 */
+	public function get_filled_job_listings(): array {
+		if ( ! get_option( 'job_manager_hide_filled_positions' ) ) {
+			return [];
+		}
+
+		$filled_jobs_transient = get_transient( 'hide_filled_jobs_transient' );
+		if ( false === $filled_jobs_transient ) {
+			$filled_jobs_transient = get_posts(
+				[
+					'post_type'      => 'job_listing',
```

Comment:
Another potential optimization here could be filtering by post status, so we don't add posts that wouldn't be displayed anyway by the main query. (Eg trashed/draft/expired? ones). Not sure if there are sites keeping around a large number of non-active listings where it could matter.

Reply:
It looks like `publish` is the default (or `inherit` if we're looking at an attachment) if no argument is passed for `post_status`.

https://github.com/WordPress/wordpress-develop/blob/bea4307daa21a3f97d159898b0ac676332e0e7de/src/wp-includes/post.php#L2397

```
$parsed_args = wp_parse_args( $args, $defaults );
if ( empty( $parsed_args['post_status'] ) ) {
	$parsed_args['post_status'] = ( 'attachment' === $parsed_args['post_type'] ) ? 'inherit' : 'publish';
}
```

However, to be safe, because you never know what sites a WPJM user might have, I've gone ahead and set `get_posts` specifically to pull only `post_status => publish`.

Committed https://github.com/Automattic/WP-Job-Manager/pull/2449/commits/fc8e201e12ce8dff5d75e59beced538f93b315c8